jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/IOResources.java

```diff
@@ -135,14 +149,11 @@ public static Content.Source asContentSource(Resource resource, ByteBufferPool b
         // Try to find an optimized content source.
         Path path = resource.getPath();
         if (path != null)
-        {
             return Content.Source.from(new ByteBufferPool.Sized(bufferPool, direct, bufferSize), path, 0, -1);
-        }
         if (resource instanceof MemoryResource memoryResource)
-        {
-            byte[] bytes = memoryResource.getBytes();
-            return new ByteBufferContentSource(ByteBuffer.wrap(bytes));
-        }
+            return new ByteBufferContentSource(ByteBuffer.wrap(memoryResource.getBytes()));
+        if (resource instanceof Content.Source.Factory factory)
```

Comment:
This check must come before calling `resource.getPath`, otherwise we cannot support `Resources` that return a `Path` for other reasons.